File: problems/basic/1/solution2.cpp
// Accepted: 11ms (70.43%), 11.27MB (35.02%)

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> targetIndices;

        for (int i = 0; i < nums.size(); ++i) {
            int num = nums[i];

            if (targetIndices.count(num) != 0) {
                int targetIndex = targetIndices[num];
                return {i, targetIndex};
            }

            targetIndices[target-num] = i;
        }

        return { -1, -1 }; // failed
    }
};

File: problems/basic/1/solution2.py
# Accepted: 54ms (95.12%), 18.83MB (7.05%)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        targetIndices: Dict[int, int] = dict()

        for i, num in enumerate(nums):
            if num in targetIndices:
                targetIndex = targetIndices[num]
                return [targetIndex, i]

            targetIndices[target-num] = i

        return [-1, -1] # failed
